kvm.py

```diff
@@ -10,50 +10,75 @@
-message = ''
+def main():
+    conn = libvirt.openReadOnly(None)
+    dom_ids = conn.listDomainsID()
+    hostname = socket.gethostname().replace('.', '_')
+    now = str(int(time.time()))
+    for dom_id in dom_ids:
+        dom = conn.lookupByID(dom_id)
+        name = dom.name().replace('.', '_')
+        total_cpu = 0
+        for vcpu in dom.vcpus()[0]:
+            cputime = vcpu[2] / 1E9
+            print(
+                'servers.{0}.virtualisation.vserver.{1}.vcpu.{2}.time {3} {4}'
+                .format(hostname, name, vcpu[0], cputime, now)
+            )
+            total_cpu += cputime
+        print(
+            'servers.{0}.virtualisation.vserver.{1}.vcpu.time {2} {3}'
+            .format(hostname, name, total_cpu, now)
+        )
+        tree = ET.fromstring(dom.XMLDesc())
+        for target in tree.findall('devices/interface/target'):
+            dev = target.attrib['dev']
+            stats = dom.interfaceStats(dev)
+            print(
+                'servers.{0}.virtualisation.vserver.{1}.net.{2}.bytesIn {3} {4}'
+                .format(hostname, name, dev, stats[0], now)
+            )
+            print(
+                'servers.{0}.virtualisation.vserver.{1}.net.{2}.bytesOut {3} {4}'
+                .format(hostname, name, dev, stats[4], now)
+            )
+            print(
+                'servers.{0}.virtualisation.vserver.{1}.net.{2}.pktsIn {3} {4}'
+                .format(hostname, name, dev, stats[1], now)
+            )
+            print(
+                'servers.{0}.virtualisation.vserver.{1}.net.{2}.pktsOut {3} {4}'
+                .format(hostname, name, dev, stats[5], now)
+            )
+        for target in tree.findall('devices/disk/target'):
+            dev = target.attrib['dev']
+            stats = dom.blockStatsFlags(dev)
+            print(
+                'servers.{0}.virtualisation.vserver.{1}.disk.{2}.bytesRead {3} {4}'
+                .format(hostname, name, dev, stats['rd_bytes'], now)
+            )
+            print(
+                'servers.{0}.virtualisation.vserver.{1}.disk.{2}.bytesWrite {3} {4}'
+                .format(hostname, name, dev, stats['wr_bytes'], now)
+            )
+            print(
+                'servers.{0}.virtualisation.vserver.{1}.disk.{2}.iopsRead {3} {4}'
+                .format(hostname, name, dev, stats['rd_operations'], now)
+            )
+            print(
+                'servers.{0}.virtualisation.vserver.{1}.disk.{2}.iopsWrite {3} {4}'
+                .format(hostname, name, dev, stats['wr_operations'], now)
+            )
+            print(
+                'servers.{0}.virtualisation.vserver.{1}.disk.{2}.ioTimeMs_read {3} {4}'
+                .format(hostname, name, dev, stats['rd_total_times'] / 1E6, now)
+            )
+            print(
+                'servers.{0}.virtualisation.vserver.{1}.disk.{2}.ioTimeMs_write {3} {4}'
+                .format(hostname, name, dev, stats['wr_total_times'] / 1E6, now)
+            )
```

Comment:
Might it be better to collect the strings in a list and print them all in a single print function in the end? I don't really know but was just wondering if it might be more performant.